MM2RandoLib/Randomizers/RText.cs

```diff
@@ -279,78 +297,88 @@ public void Randomize(Patch p, Random r)
                     {
                         newIndex = m;
                         break;
                     }
                 }
 
-                for (int j = 0; j < 9; j++)
+                for (Int32 j = 0; j < 9; j++)
                 {
-                    int dmg = RWeaknesses.BotWeaknesses[newIndex, j];
-                    sb.Append($"{GetBossWeaknessDamageChar(dmg)} ");
+                    Int32 dmg = RWeaknesses.BotWeaknesses[newIndex, j];
+                    sb.Append($"{RText.GetBossWeaknessDamageChar(dmg)} ");
                 }
 
-                string rowString = sb.ToString().Trim();
-                for (int j = 0; j < rowString.Length; j++)
+                String rowString = sb.ToString().Trim();
+
+                for (Int32 j = 0; j < rowString.Length; j++)
                 {
-                    p.Add(txtRobos[i] + j,
-                        CreditsCipher[rowString[j]],
+                    in_Patch.Add(txtRobos[i] + j,
+                        rowString[j].AsCreditsCharacter(),
                         $"Credits robo weakness table char #{j + i * rowString.Length}");
                 }
             }
 
             // Write Wily Boss damage table
-            for (int i = 0; i < txtWilys.Length; i++)
+            for (Int32 i = 0; i < txtWilys.Length; i++)
             {
                 sb = new StringBuilder();
-                for (int j = 0; j < 8; j++)
+
+                for (Int32 j = 0; j < 8; j++)
                 {
-                    int dmg = RWeaknesses.WilyWeaknesses[i, j];
-                    sb.Append($"{GetBossWeaknessDamageChar(dmg)} ");
+                    Int32 dmg = RWeaknesses.WilyWeaknesses[i, j];
+                    sb.Append($"{RText.GetBossWeaknessDamageChar(dmg)} ");
                 }
 
                 sb.Remove(sb.Length - 1, 1);
-                string rowString = sb.ToString();
+                String rowString = sb.ToString();
 
-                for (int j = 0; j < rowString.Length; j++)
+                for (Int32 j = 0; j < rowString.Length; j++)
                 {
-                    p.Add(txtWilys[i] + j,
-                        CreditsCipher[rowString[j]],
+                    in_Patch.Add(txtWilys[i] + j,
+                        rowString[j].AsCreditsCharacter(),
                         $"Credits wily weakness table char #{j + i * rowString.Length}");
                 }
             }
         }
 
-        public void FixWeaponLetters(Patch p, int[] permutation)
+
+
+        public void FixWeaponLetters(Patch in_Patch, Int32[] in_Permutation)
         {
+            const Int32 WEAPON_GET_LETTERS_ADDRESS = 0x037E22;
+
             // Re-order the letters array to match the ordering of the shuffled weapons
-            char[] newLettersPermutation = new char[9];
-            newLettersPermutation[0] = newWeaponLetters[0];
-            for (int i = 0; i < 8; i++)
+            Char[] newLettersPermutation = new Char[9];
```

Comment:
I just realized I have the same question about `char` and `Char`. With `int` vs. `Int32` I thought it was just about being explicit about bitwidth but now I'm not so certain.

Reply:
I like to be clear and consistent, so I prefer using the Object types, rather than the built-in types.

Reply:
There is no different functionally, EXCEPT when you use enum vs Enum.  Those are different, and it makes me upset that C# designers did that, so, I always like to use Object types in my coding.